### PR TITLE
feat(reasoner): support value type inference in udf

### PR DESCRIPTION
**Pull request by fishjoy**

### CODE REVIEW

File: reasoner/udf/src/main/java/com/antgroup/openspg/reasoner/udf/UdfMng.java [hunk 1]
```diff
@@ -19,10 +19,11 @@
 import com.antgroup.openspg.reasoner.udf.model.UdafMeta;
 import com.antgroup.openspg.reasoner.udf.model.UdfParameterTypeHint;
 import com.antgroup.openspg.reasoner.udf.model.UdtfMeta;
+import java.io.Serializable;
 import java.util.List;
 import java.util.Map;
 
-public interface UdfMng {
+public interface UdfMng extends Serializable {
```
Comment (youdonghai) [inline reasoner/udf/src/main/java/com/antgroup/openspg/reasoner/udf/UdfMng.java:26]: This class is not allowed to be serialized as it contains a Java Method Object internally.
you can call UdfMngFactory.getUdfMng to retrieve an instance

**Review by youdonghai — CHANGES_REQUESTED** [review]

File: reasoner/udf/src/main/java/com/antgroup/openspg/reasoner/udf/UdfMng.java (hunk 1 above)
Comment (fishjoy) [reply reasoner/udf/src/main/java/com/antgroup/openspg/reasoner/udf/UdfMng.java]: fixed

> This class is not allowed to be serialized as it contains a Java Method Object internally. you can call UdfMngFactory.getUdfMng to retrieve an instance

**Review by youdonghai — APPROVED** [review]
LGTM

**Review by royzhao — APPROVED** [review]
LGTM

**Pull request merged**